rasa/core/story_conflict.py

```diff
@@ -0,0 +1,252 @@
+        """
+        Iterate over all given trackers and all sliced states within
+        each tracker, where the slicing is based on `max_history`
+        :param trackers: List of trackers
+        :param domain: Domain (used for tracker.past_states)
+        :param max_history: Assumed `max_history` value for slicing
+        :return: Yields (tracker, event, sliced_states) triplet
+        """
+        for tracker in trackers:
+            states = tracker.past_states(domain)
+            states = [
+                dict(state) for state in states
+            ]  # From rasa/core/featurizers.py:318
+
+            idx = 0
+            for event in tracker.events:
+                if isinstance(event, ActionExecuted):
+                    sliced_states = MaxHistoryTrackerFeaturizer.slice_state_history(
+                        states[: idx + 1], max_history
+                    )
+                    yield tracker, event, sliced_states
+                    idx += 1
+
+    @staticmethod
+    def _get_prev_event(
+        state: Optional[Dict[Text, float]]
+    ) -> [Optional[Text], Optional[Text]]:
+        """
+        Returns the type and name of the event (action or intent) previous to the
+        given state
+        :param state: Element of sliced states
+        :return: (type, name) strings of the prior event
+        """
+        prev_event_type = None
+        prev_event_name = None
+
+        if not state:
+            return prev_event_type, prev_event_name
+
+        for k in state:
+            if (
+                k.startswith(PREV_PREFIX)
+                and k[len(PREV_PREFIX) :] != ACTION_LISTEN_NAME
+            ):
+                prev_event_type = "action"
+                prev_event_name = k[len(PREV_PREFIX) :]
+
+            if not prev_event_type and k.startswith(MESSAGE_INTENT_ATTRIBUTE + "_"):
+                prev_event_type = "intent"
+                prev_event_name = k[len(MESSAGE_INTENT_ATTRIBUTE + "_") :]
+
+        return prev_event_type, prev_event_name
+
+    def add_conflicting_action(self, action: Text, story_name: Text):
+        """
+        Add another action that follows from the same state
+        :param action: Name of the action
+        :param story_name: Name of the story where this action
+        is chosen
+        """
+        if action not in self._conflicting_actions:
+            self._conflicting_actions[action] = [story_name]
+        else:
+            self._conflicting_actions[action] += [story_name]
+
+    @property
+    def conflicting_actions(self) -> List[Text]:
+        """
+        Returns the list of conflicting actions
+        """
+        return list(self._conflicting_actions.keys())
+
+    @property
+    def conflicting_actions_with_counts(self) -> List[Text]:
+        """
+        Returns a list of strings, describing what action
+        occurs how often
+        """
+        return [f"{a} [{len(s)}x]" for (a, s) in self._conflicting_actions.items()]
```

Comment:
Can we please give `a` and `s` speaking names?
```suggestion
        return [f"{a} [{len(s)}x]" for (a, s) in self._conflicting_actions.items()]
```